### Problem

gcr から取得した dockerコンテナを起動しようとすると

```
docker run --gpus all -p 8080:8080 -v `pwd`:/kaggle -d --name kaggle --restart=always  gcr.io/kaggle-gpu-images/python:v100
 ```
 
下記のようなエラーが表示される
 
(HTTP code 500) server error - OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: mount error: file creation failed: /var/lib/docker/overlay2/6c21b9fd2e001784030fee74c54ee5b08e9ba7a516afc3109e46fd7bd77c0cad/merged/usr/lib/x86_64-linux-gnu/libnvidia-ml.so.1: file exists: unknown

### dockerコンテナ起動しない問題の対処

cuda driver 470.76 を入れてみた -> 治らず
https://github.com/NVIDIA/nvidia-docker/issues/1496

gcrにあるイメージではなく自分でbuildしてみる -> docker runは成功するが, 一部のnotebookの処理で失敗
https://github.com/Kaggle/docker-python

### gpu kernelの問題対処

いつもの gpuカーネルのテストを回してみると、
https://www.kaggle.com/yuwenwen/running-kaggle-kernels-with-a-gpu

```
my_model.fit_generator(train_generator, epochs=5, validation_data=val_generator)
```
の部分で `NotFoundError: No algorithm worked! when using Conv2D` といってエラーとなる

下記のIssueにある対処を冒頭に入れると問題が発生しなくなった
https://github.com/tensorflow/tensorflow/issues/43174#issuecomment-691657692

```
from tensorflow.compat.v1 import ConfigProto
from tensorflow.compat.v1 import InteractiveSession

config = ConfigProto()
config.gpu_options.allow_growth = True
session = InteractiveSession(config=config)
```

